drivers/shtc1/shtc1.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2017, RWTH-Aachen, Steffen Robertz, Josua Arndt
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include "shtc1.h"
+
+#define ENABLE_DEBUG	(0)
+#include "debug.h"
+
+
+/*
+ * does a crc check and returns 0 for passed and -1 for failed
+ */
+int8_t _check_crc(uint8_t *rec_values, uint8_t right_crc)
+{
+	uint8_t crc = 0xFF;
+	uint8_t current_byte;
+	uint8_t bit;
+
+	for (current_byte = 0; current_byte < 2; current_byte++)
+	{
+		crc ^= (rec_values[current_byte]);
+		for (bit = 8; bit > 0; bit--)
+		{
+			if (crc & 0x80)
+				crc = (crc << 1) ^ 0x31;
+			else
+				crc = (crc << 1);
+		}
+	}
+	return !(crc == right_crc);
+}
+
+int8_t shtc1_init(i2c_t dev)
+{
+	int8_t error = i2c_init_master(dev, I2C_SPEED_FAST);
+	if(!error) {
+		 return 0;
+	}else {
+		return error;
+	}
+}
+
+int8_t shtc1_measure(i2c_t dev, crc_type_t crc, shtc1_values_t* received_values)
+{
+	uint8_t data[] = { 0x7C, 0xA2 };
+	i2c_write_bytes(dev, 0x70, data, 2);
+	uint8_t received[6];
+	i2c_read_bytes(dev, 0x70, received, 6);
+	uint16_t temp_f = ((received[0]<<8)|received[1]);
+	uint16_t abs_humidity = ((received[3]<<8)|received[4]);
+	if(crc){
+		if(!((_check_crc(&received[0], received[2]) == 0) && (_check_crc(&received[3], received[5]) == 0))){
+			/* crc check failed */
+			DEBUG("CRC Error");
+			return -1;
+		}
+		DEBUG("CRC Passed! \n");
+	}
+	received_values->temp = (175.0*temp_f/65536)-45;
+	received_values->rel_humidity = 100*(abs_humidity/65536.0);
+
```

Comment:
Usually we abstain from extra divisions and normalization operations to represent a value in persent or something. Do you think we can simplify here?

Reply:
I believe that a conversion of the temperature to °C is neccessary. The raw values does not hold a lot of information to a user.
The relative humidity can be changed back to the absolute one as this should be clear.

Reply:
Let's see at the driver guide 

[ On top, devices should work with physical values where ever possible. So e.g. sensors should return already converted values in some physical unit instead of RAW data, so that users can work directly with data from different devices directly without having to care for device specific conversion.](https://github.com/RIOT-OS/RIOT/wiki/Guide:-Writing-a-device-driver-in-RIOT)